src/Integration/MefServices/ActiveSolutionBoundTracker.cs

```diff
@@ -146,20 +151,26 @@ private async Task UpdateConnectionAsync(BindingConfiguration bindingConfigurati
             Debug.Assert(!sonarQubeService.IsConnected,
                 "SonarQube service should always be disconnected at this point");
 
-            var boundProject = bindingConfiguration.Project;
-
-            if (boundProject != null)
+            if (!bindingConfiguration.Mode.IsInAConnectedMode())
             {
-                var connectionInformation = boundProject.CreateConnectionInformation();
-                await WebServiceHelper.SafeServiceCallAsync(() => sonarQubeService.ConnectAsync(connectionInformation,
-                    CancellationToken.None), this.logger);
+                return true;
```

Comment:
This return feels very awckard inside this method. 
Why not having it in the caller method like so:

` if (!newBindingConfiguration.Mode.IsInAConnectedMode())
 {
     var updateConnection = await UpdateConnectionAsync(newBindingConfiguration);
     if (!updateConnection)
     {
         CurrentConfiguration = BindingConfiguration.Standalone;
         return;
     }
 }
`

We would still probably have to call the disconnect SonarQube

Reply:
It is closer to the context of the UpdateConnectionAsync method, it is part of updating the connection.

Reply:
It doesn't feel so for me. Because we are saying that "yes" the connection was updated, although it is in standalone. Feels very, very confusing. 
My proposal made it more explicit when we should connect.